Issue #842: Update Bartik color and tab settings

docwilmot opened the pull request:
For https://github.com/backdrop/backdrop-issues/issues/842
Updated for 1.2.2

docwilmot commented:
Force push without my settings.php again :anguished:

jenlampton commented on core/themes/bartik/color/color.inc:
```diff
@@ -15,14 +15,16 @@ $info = array(
   'schemes' => array(
     'default' => array(
-      'title' => t('Blue Lagoon (default)'),
+      'title' => t('Blue Drop (default)'),
```
Blue Lagoon here is fine.

docwilmot replied:
For this PR the default color is Blue Drop. Are we changing to black? We decided to keep Blue Lagoon (the old Bartik gradient) for legacy reasons, so I chose "Blue Drop" (flat blue) to distinguish the two blues.

If we have decided on black as default, I think "Black Drop" sounds nice (a play on Back Drop) and although it doesn't jive with your minimalist preference I think a more quirky title matches the one its replacing (Blue lagoon), and the other schemes, none of which are colors, but schemes (Mocha, Giant Goldfish, Ice).

jenlampton replied:
We should keep the name "Blue Lagoon" - but change it's colors. If you really like the name Black Drop we can keep it :)

jenlampton commented on core/themes/bartik/color/color.inc:
```diff
@@ -86,6 +94,56 @@ $info = array(
+        'activemenu' => '#4c5037',
+      ),
+    ),
+    'black_drop' => array(
```
What's the difference between Black Drop and the default? Let's remove one to be less confusing.

docwilmot replied:
The default is blue. Black drop is an option.

jenlampton commented on core/themes/bartik/color/preview.html:
```diff
@@ -5,7 +5,7 @@
       <ul id="preview-main-menu-links">
         <li><a>Home</a></li>
-        <li><a>Te Quidne</a></li>
+        <li><a class="active">Linkus Activus</a></li>
```
cute!

jenlampton commented on core/themes/bartik/css/maintenance-page.css:
```diff
@@ -48,7 +48,7 @@ body.maintenance-page {
   font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
 }
-.maintenance-page .page-title {
+.maintenance-page  h1#page-title {
```
nice catch

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -551,7 +551,7 @@ ul.tips {
   float: left; /* LTR */
   list-style: none;
-  padding: 0 1px;
+  /* padding: 0 1px; */
```
let's delete this line if we don't need it

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -587,6 +598,24 @@ ul.tips {
 .l-header ul.menu li a.active {
   border-bottom: none;
 }
+.no-tabs .l-header ul.menu a {
```
Let's exclude all the tabs changes.

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -566,7 +566,18 @@ ul.tips {
+  color: #333;
+  background: #f6f6f2;
+  background: rgba(255, 255, 255, 0.95);
+}
```
Let's exclude all the tabs changes.

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -1073,20 +1101,37 @@ article .link-wrapper ul.links {
+}
+.rounded-tabs .tabs ul.primary li a {
+  background-color: #ededed;
+  padding: 0 10px 3px;
```
Let's exclude all the tabs changes.

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -1073,20 +1101,37 @@ article .link-wrapper ul.links {
+}
+.no-tabs ul.primary li.active a {
+  border:none;
+}
```
Let's exclude all the tabs changes.

jenlampton commented on core/themes/bartik/template.php:
```diff
@@ -8,6 +8,13 @@ function bartik_preprocess_maintenance_page(&$variables) {
+function bartik_preprocess_page(&$variables) {
+  $variables['classes'][] = theme_get_setting('main_menu_tabs', 'bartik');
+}
+
```
Let's exclude all the tabs changes.

jenlampton commented on core/themes/bartik/templates/node.tpl.php:
```diff
@@ -77,9 +77,8 @@
  */
 ?>
-<article id="node-<?php print $node->nid; ?>" class="<?php print implode(' ', $classes); ?> clearfix"<?php print backdrop_attributes($attributes); ?>>
+<article id="node-<?php print $node->nid; ?>" class="<?php print implode(' ', $classes); ?> clearfix"<?php print backdrop_attributes($attributes); ?> role="article">
```
Was this change intentional?

docwilmot replied:
Perhaps, I don't actually recall, embarrassingly. If it shouldn't be so, we can kill it.

jenlampton commented on core/themes/bartik/theme-settings.php:
```diff
@@ -10,3 +10,16 @@
 $form['color'] = array(
   '#markup' => '<p>' . t('This theme supports custom color palettes if the Color module is enabled on the <a href="!url">modules page</a>. Enable the Color module to customize this theme.', array('!url' => url('admin/modules'))) . '</p>',
 );
+$form['tabs_wrapper'] = array(
```
Let's exclude all the tabs changes.

jenlampton commented on core/themes/bartik/bartik.info:
```diff
@@ -11,4 +11,5 @@ stylesheets[print][] = css/print.css
 ; We need at least one setting for the theme settings page to appear in the
 ; menu. Color module provides our defaults, so we just register a dummy setting.
 settings[color] = true
+settings[main_menu_tabs] = no-tabs
```
Let's exclude all the tabs changes.

docwilmot commented:
> Let's exclude all the tabs changes.

This was meant to allow people who wanted to preserve the rounded tabs to do so. I personally think there won't be more than 1 person who wants those tabs, so happy to nix all the tabs stuff in the PR if you wish. 

But what do we replace the rounded tabs with? I had put in options for rounded tabs, square tabs or no tabs (just flat links for the main menu and grey squares for the local links). Do we stick with round, or square or no tabs?

jenlampton commented:
I just talked it over with @quicksketch and he filled me in on why we need the tab thing to be a setting (because we want to remove the rounded-ness, but we can't go changing the way people's sites look if they are actually using Bartik). So I get it now! Let's leave the tabs, you did a lot of good work here :)

docwilmot commented:
@jenlampton can I get some eyes on this please? It still needs work as the tests are failing, but should be quick fix. 

However, I need a look at the approach to the upgrade path. It is a bit complex for a theme that is likely  **not** being used by many people in production. 

I assumed that if we just changed 'default' to Blue Lagoon' as you suggested, this would result in a change immediately in people's theme appearance so it was best to find a way to preserve current palette settings. Furthermore, there isn't a simple way to do that change you suggested anyway. So opted to save whatever current color palette a site is using to 'custom' on upgrade. Have a look, let me know please, so I can fix this and ship it for 1.3.0.

jenlampton commented:
Okay, I'll take a look. I do think we need to change the default (for new sites) so they are not blue. We need to move away from the Drupal look-and-feel as much as possible :) I'll see if I can figure out an upgrade path. 

It should be something like... 1) load config. 2) change theme setting from 'default' to 'Blue lagoon' (keeping current colors). 3) save config.

docwilmot commented:
> It should be something like... 1) load config. 2) change theme setting from 'default' to 'Blue lagoon' (keeping current colors). 3) save config.

Let me save you some headache: that's not going to be it! When you load the config file you'll find that it looks like this:

```
    "color": {
        "palette": {
            "top": "#4c1c58",
            "bottom": "#593662",
            "bg": "#fffdf7",
            "sidebar": "#edede7",
            "sidebarborders": "#e7e7e7",
            "footer": "#2c2c28",
            "titleslogan": "#ffffff",
            "text": "#301313",
            "link": "#9d408d"
        },
        "stylesheets": [
            "public://color/bartik-c423e789/colors.css"
        ],
        "files": [
            "public://color/bartik-c423e789/colors.css"
        ]
    }
```

Note nothing about 'default' or 'Blue Lagoon'. When the admin form on the theme settings page loads the config file, Color module actually compares the palette in the config to the palettes in `color.inc`, and only then tells the admin form if it matches a palette; if not, it displays on the form as 'custom'. As a matter of fact, until you first hit "Save theme settings", the config just says `"color": NULL`.

Note also the file path for the generated CSS file is saved in the config file. So if you want to do anything programmatically, you first _must_ generate the CSS file from the palette, _then_ generate a folder name (from the MD5 of the palette) and save the file in that folder, _then_ write the path to the config _then_ save the config.

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -307,9 +307,9 @@ ul.tips {
-.admin-bar body .layout {
-  padding-top: 33px;
-  margin-top: -33px;
+body.admin-bar .layout {
```
Are we just fixing a previous mistake here?  if so then :+1:

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -505,8 +505,8 @@ ul.tips {
 }
-.l-header .block-user-login .form-submit:hover,
-.l-header .block-user-login .form-submit:focus {
+.l-header .block-user-login input.form-submit:hover,
```
why are we adding additional selectors here?

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -606,7 +634,7 @@ ul.tips {
   margin-top: 10px;
 }
-.page-title {
+h1#page-title {
```
don't like this one.

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -1161,7 +1205,7 @@ ul.action-links li a {
 /* ---------------- Buttons    ---------------- */
 
-.form-submit,
+input.form-submit,
```
additional selector.

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -1214,18 +1258,25 @@ fieldset {
 .filter-wrapper {
   top: 0;
-  padding: .5em 0;
+  padding: 1em 0 0.2em;
```
filter css updates: why the change of padding?

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -1214,18 +1258,25 @@ fieldset {
 }
-.filter-wrapper .fieldset-wrapper {
-  margin-top: 0;
+.filter-help a {
```
filter css updates in here too, were these intentional?

jenlampton commented on core/themes/bartik/templates/maintenance-page.tpl.php:
```diff
@@ -45,7 +45,7 @@
     <main id="content" class="column" role="main">
       <a id="main-content"></a>
-      <?php if ($title): ?><h1 class="page-title"><?php print $title; ?></h1><?php endif; ?>
+      <?php if ($title): ?><h1 class="title" id="page-title"><?php print $title; ?></h1><?php endif; ?>
```
don't think we want this. Markup changes?

jenlampton commented on core/themes/bartik/templates/node.tpl.php:
```diff
@@ -25,7 +25,7 @@
  *   - node: The current template type; for example, "theming hook".
  *   - node-[type]: The current node type. For example, if the node is a
- *     "Post" it would result in "node-post". Note that the machine
+ *     "Article" it would result in "node-article". Note that the machine
```
love that we are fixing these :)

jenlampton commented on core/themes/bartik/templates/node.tpl.php:
```diff
@@ -119,7 +117,7 @@
   <?php endif; ?>
 
-  <?php if ($page && isset($comments['comments'])): ?>
+  <?php if ($page && !empty($comments)): ?>
```
this change will detect both the comments and the comment form. Previously we were just looking for comments. Let's be consistent with what's happening in /node/templates/node.tpl.php unless we're intentionally doing something different in bartik.

jenlampton commented:
> Let me save you some headache

Thank you :)

> Color module actually compares the palette in the config to the palettes in color.inc, and only then tells the admin form if it matches a palette; if not, it displays on the form as 'custom'. 

That might work out to be even better.  Then we can change the colors all we want and not risk anyone loosing their settings, as long as we can result in the old colors being saved as custom :)

I'm running through some tests now to see what happens in config and UI before and after this change (so far, I've only done before). And we'll see how crazy that update gets :) Thanks for taking a stab at it!

I'm going to rebase and push up a new PR with some of these changes (perhaps accidental?) omitted.

docwilmot commented:
This PR was done centuries ago; I just added the update hook recently and rebased. I don't recall how these got in there, I should have checked though. Excuse the extra work, sorry.

jenlampton commented on core/themes/bartik/css/colors.css:
```diff
@@ -7,45 +7,62 @@ body {
   border-color: #ffffff;
 }
-.l-container,
+.l-container, .layout {
```
different selectors go on new lines:

```
.l-container,
.layout {
```

jenlampton commented on core/themes/bartik/css/colors.css:
```diff
@@ -7,45 +7,62 @@ body {
 .tabs ul.primary li.active a {
+  background-color: #e8e6e5;
+}
+.rounded-tabs .tabs ul.primary li.active a, .square-tabs .tabs ul.primary li.active a {
```
new lines :)

jenlampton commented on core/themes/bartik/css/style.css:
```diff
@@ -1214,18 +1258,25 @@ fieldset {
-  margin-top: 0;
+.filter-help a {
+  font-size: 0.857em;
+  padding: 2px 20px 0;
```
I think we removed filter help in a recent commit.

jenlampton commented:
> Excuse the extra work, sorry.

It's no problem, you're just honing my Git skills! :)  Thanks for all the fantastic work on this so far!

jenlampton commented on core/modules/system/system.install:
```diff
@@ -2650,6 +2650,97 @@ function system_update_1045() {
+  global $theme;
+
+  // Only bother if the current theme is Bartik.
+  if ($theme == 'bartik') {
```
when update.php is running the active theme is `seven`. We need to check the default theme instead.

docwilmot commented:
`Undefined index: base` error happens when there is a color in the `color.inc` file that isn't in the palette. I've just realised that my final rebase was what caused all those stray CSS issues and the Git confliccts. I had these errors sorted including the default theme check. Blast! I seem to have rolled back my branch to a very old state somehow!

jenlampton commented:
@docwilmot ohnoes!  You will be pleased to learn that your headache has indeed saved me a headache.  I've been looking at this weird `base` problem as well as a few others, and unfortunately (or maybe fortunately?) we've decided to take another approach for preserving the legacy blue lagoon colors.

See, we forgot to account for the use-case where people are using the bartik theme (original Blue Lagoon color palette) but are not using the color module.  This approach with using color module to preserve those legacy colors won't work for them.

So what we'd like to do instead is include a legacy color style sheet in bartik, and in the update hook all we need to do is set a flag saying when it will need to be used.

docwilmot commented:
I considered that approach too. But also consider the case of users who are using any palettes we're dropping (firehouse and slate, which were ugly and too similar to Black Drop respectively) and palettes we've changed (all the rest).

jenlampton commented:
Anyone who has used any palette in the past (other than default) will have the palette saved in config. Loosing the name doesn't change any of their settings, as the settings will remain - so the style sheets will remain and don't need to be regenerated.  The only difference for those people is that the name will appear to change from "Firehouse" to "Custom".

Basically anyone who has used color module in the past will be fine.  It's only the default use-case we need to support.

jenlampton commented:
Things are coming along nicely over at https://github.com/backdrop/backdrop/pull/1129

quicksketch commented:
@jenlampton rerolled this patch at https://github.com/backdrop/backdrop/pull/1129 and then I rerolled that one at https://github.com/backdrop/backdrop/pull/1194.
